doc/doxygen/src/move-semantics.md

```diff
@@ -0,0 +1,76 @@
+Semantics of typedefs and moving their values {#typedef-and-move}
+=============================================
+
+This page describes general properties of types defined (`typedef`'d) in RIOT.
+It is mostly aimed at developers who alter or extend RIOT;
+API users can often ignore them if they stick to idiomatic C.
+
+This is being documented for consistency within RIOT,
+and for the benefit of users who use RIOT
+from other languages (such as @rust using-rust "Rust" )
+or do other advanced things like implementing coroutines.
+
+Movability
+----------
+
```

Comment:
This is a nice explanation, but also slightly verbose and might be missing the "why do we care?".

Something along the lines of:

```
The C compiler in general doesn't move around data that has been allocated on stack or on the heap by itself, so movability is usually not something that ideomatic C needs to deal with.

The Rust compiler does move data around, if it deems it beneficial for optimization.

As most of RIOT's C APIs are being used by Rust through wrappers, movability of a type needs to be specified in order to no force Rust to assume *may never move* on *all* types.
```

Reply:
Thanks, taken in and modified (because Rust doesn't just move data around either).